tests/test_vagrant_source.py

```diff
@@ -19,6 +19,53 @@
 from hostsresolver import vagrant_source
 
 
+class TestFindVagrantRoot(unittest.TestCase):
+    def setUp(self):
+        self.origin_vagrant_dotfile_path = vagrant_source.VAGRANT_DOTFILE_PATH
+        self.origin_vagrant_vagrantfile = vagrant_source.VAGRANT_VAGRANTFILE
+        self.origin_vagrant_cwd = vagrant_source.VAGRANT_CWD
+        self.addCleanup(self.cleanupVagrantVariables)
+
+        self.valid_vagrant_root = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'vagrant_project')
```

Comment:
maybe make "os.path.join(os.path.dirname(os.path.abspath(__file__)), ????)" a function, it appears at least 6 times in this file only